usim/_primitives/timing.py

```diff
@@ -265,6 +265,7 @@ class Delay(Notification):
     __slots__ = ('duration',)
 
     def __init__(self, duration: float):
+        assert duration > 0, "delay must point at the future"
```

Comment:
This means I still cannot do `time + 0`. Isn't this contradictory to the goal of this PR?

Reply:
Given implementation of `Time.__add__` below it is not. But then the line in the documentation is misleading "The expression ``time + duration`` is equivalent to ``Delay(duration)``.". Whenever ``duration == 0`` it would then be equivalent to ``time + instant``.

Reply:
I've changed extended the description a bit to describe how ``Delay`` relates to ``time`` and the other types. Pleas have another look.